pkg/csi/service/wcpguest/controller_helper.go

```diff
@@ -186,7 +186,7 @@ func checkPVCCondition(ctx context.Context, pvc *v1.PersistentVolumeClaim,
 			pvc.Namespace, condition.Type)
 		if condition.Type == reqCondition {
 			pvcSize := pvc.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
-			if pvcSize == *reqSize {
+			if pvcSize.Size() == reqSize.Size() {
```

Comment:
I have used `pvcSize.Value()` in the quota framework, basically, `.Value()` is reliable comparison. With your changes above, I think comparing Gi to Mi will fail even if they are exactly same.